src/modules/vision_target_estimator/Orientation/VTEOrientation.cpp

```diff
@@ -0,0 +1,345 @@
+		target_innov.innovation_variance = _target_estimator_orientation->computeInnovCov(
+				target_orientation_obs.meas_unc_theta);
+		target_innov.innovation = _target_estimator_orientation->computeInnov(target_orientation_obs.meas_theta);
+		// Set the Normalized Innovation Squared (NIS) check threshold. Used to reject outlier measurements
+		_target_estimator_orientation->setNISthreshold(_nis_threshold);
+		meas_fused = _target_estimator_orientation->update();
+
+		// Fill the target innovation field
+		target_innov.innovation_rejected = !meas_fused;
+		target_innov.fused = meas_fused;
+
+		target_innov.observation = target_orientation_obs.meas_theta;
+		target_innov.observation_variance = target_orientation_obs.meas_unc_theta;
+		target_innov.timestamp_sample = target_orientation_obs.timestamp;
+		target_innov.timestamp = hrt_absolute_time();
+
+		// log test ratio defined as _innov / _innov_cov * _innov. If test_ratio > _nis_threshold, no fusion
+		target_innov.test_ratio = _target_estimator_orientation->getTestRatio();
+
+	} else {
+		// No yaw measurement
+		target_innov.fused = false;
+	}
+
+	_vte_aid_ev_yaw_pub.publish(target_innov);
+
+	return meas_fused;
+}
+
+
+void VTEOrientation::publishTarget()
+{
+	vision_target_est_orientation_s vision_target_orientation{};
+
+	vision_target_orientation.timestamp = _last_predict;
+	vision_target_orientation.orientation_valid = (hrt_absolute_time() - _last_update < target_valid_TIMEOUT_US);
+
+	vision_target_orientation.theta = _target_estimator_orientation->getPosition();
+	vision_target_orientation.cov_theta =  _target_estimator_orientation->getPosVar();
+
+	if (_target_mode == TargetMode::Moving) {
+		vision_target_orientation.v_theta = _target_estimator_orientation->getVelocity();
+		vision_target_orientation.cov_v_theta =  _target_estimator_orientation->getVelVar();
+	}
+
+	_targetOrientationPub.publish(vision_target_orientation);
+}
+
+void VTEOrientation::_check_params(const bool force)
+{
+	if (_parameter_update_sub.updated() || force) {
+		parameter_update_s pupdate;
+		_parameter_update_sub.copy(&pupdate);
+
+		updateParams();
+	}
+
+	if (_range_sensor.valid) {
+		_range_sensor.valid = (hrt_absolute_time() - _range_sensor.last_update) < measurement_updated_TIMEOUT_US;
+	}
+}
+
+void VTEOrientation::updateParams()
+{
+
+	ModuleParams::updateParams();
+
+	_yaw_unc = _param_vte_yaw_unc_in.get();
+	_ev_angle_noise = _param_vte_ev_angle_noise.get();
+	_ev_noise_md = _param_vte_ev_noise_md.get();
+	_nis_threshold = _param_vte_yaw_nis_thre.get();
+}
+
+void VTEOrientation::set_range_sensor(const float dist, const bool valid)
+{
+	_range_sensor.valid = valid;
+	_range_sensor.dist_bottom = dist;
+	_range_sensor.last_update = hrt_absolute_time();
+}
+
+bool VTEOrientation::selectTargetEstimator()
+{
+	Base_KF_orientation *tmp_theta = nullptr;
+
+	bool init_failed = true;
+
+	if (_target_mode == TargetMode::Moving) {
+		tmp_theta = new KF_orientation_moving;
```

Comment:
Are we sure that this will never run after arming?

Reply:
`bool VTEPosition::selectTargetEstimator()` is only called in `bool VTEPosition::init()` which is called in  _VisionTargetEst.cpp_ in the method `bool VisionTargetEst::init()` (`_vte_position->init()`) which is only called in the task spawn: `int VisionTargetEst::task_spawn(int argc, char *argv[])`

Unless the task is manually restarted after arming, it should not run.